Implement replying to irc messages in private

richo opened the pull request:
Hi!

I'm writing an irc game that requires sharing some state with the user in private.

This probably can't be merged as is (unless you're ok with being slightly broken on master) since it'll crash slack. I wanted to see if you had feedback before I go work out how to make this work for the slack adaptor.

richo committed:
Implement replying to irc messages in private

jwilm commented:
Hi @richo!

Thanks for the PR :smile:. I don't see any problem with merging this without slack support as long as the `AdapterMsg::Private` is handled in the slack loop (can be a noop, we just don't want it to reach the `_` case). We can just open an issue for adding PM support to slack.

For the feature, I am wondering whether the `PrivateMessage` type is really needed. It seems like all of the data you need is already on `OutgoingMessage`, and the `reply_private` method on `IncomingMessage` could just stuff it into the `AdapterMsg::Private` variant. What do you think?

richo committed:
Drop slack private messages on the floor for now

richo committed:
Handle Shutdown messages

This means that adding code that would break code will refuse to compile

richo commented:
Hello!

Ok I updated the PR. I nopped out pm's in slack (with a log message explaining) and unified the AdapterMessage types around OutgoingMessage.

I also implemented the Shutdown type and removed the catchall branch, so that breaking things as I tried to will refuse to compile in future :)

jwilm commented:
@richo this looks great! Thanks for removing the catchall - I :heart: fail early. I'll add slack support for this and get a release up on crates.io.
